### ISSUE TITLE
Safari on iOS/iPadOS unable to use ‘Open Files’ and ‘Load Trajectory’ modules

### ISSUE THREAD

**Issue by RowenaZireael** [molstar/molstar#1126]
When clicking ‘Select a file’ in safari, files with  .gro, .pdb .xtc .nc .prmtop .nc cannot be selected, so the trajectory cannot be loaded.

**Comment by dsehnal** [comment]
These file types are included in the in the filters for the dialog (showing just the "select model file part".

![image](https://github.com/molstar/molstar/assets/21145776/14d7777c-059e-4ec5-8671-ce5d797cca10)

Could a be a security permissions issue on your device?

**Comment by RowenaZireael** [comment]
> These file types are included in the in the filters for the dialog (showing just the "select model file part".
> 
> ![image](https://private-user-images.githubusercontent.com/21145776/331795271-14d7777c-059e-4ec5-8671-ce5d797cca10.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTY2ODI4NDYsIm5iZiI6MTcxNjY4MjU0NiwicGF0aCI6Ii8yMTE0NTc3Ni8zMzE3OTUyNzEtMTRkNzc3N2MtMDU5ZS00ZWM1LTg2NzEtY2U1ZDc5N2NjYTEwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MjYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTI2VDAwMTU0NlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWRiNzJhOWUwZmJlMTUyOTgzZDU2N2Y4NDNlN2Y2YmYwZjVhOGVmYTBkMGIyYzdiNGNjMTgzY2M3ODJhNTJkNDMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.4esj_2oj15OQHFkd3WdPegGbL18rzr9lMeoLifSteO4)
> 
> Could a be a security permissions issue on your device?



> These file types are included in the in the filters for the dialog (showing just the "select model file part".
> 
> ![image](https://private-user-images.githubusercontent.com/21145776/331795271-14d7777c-059e-4ec5-8671-ce5d797cca10.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTY2ODI4NzksIm5iZiI6MTcxNjY4MjU3OSwicGF0aCI6Ii8yMTE0NTc3Ni8zMzE3OTUyNzEtMTRkNzc3N2MtMDU5ZS00ZWM1LTg2NzEtY2U1ZDc5N2NjYTEwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MjYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTI2VDAwMTYxOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTA3NDlhNDNjMTQyOGMxNjg5NjQyMTA4OGRmM2QwYzA3ZjVhMDRlNmQ3ZWZhNDg4MWI3YjM4M2I2Y2QxODNkMzkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0._ght7_1Qgt9ArKAB4xz24_tUJwGvGCGGEoXivxMfS1s)
> 
> Could a be a security permissions issue on your device?

I don’t think so. If it was a security issue, pdb formatted files wouldn’t be able to be uploaded to other websites, such as “CHARMM-GUI”. Meanwhile interestingly, only .zip formatted files can be selected in the “Open files” module.  This has been tested on iPad Pro M1, iPad Pro M4, iPhone 7 and iPhone 13 with system version iOS/iPadOS16-17, and the issue can be reproduced. It would be nice if this could be solved, to allow trajectory analysis on mobile devices.

**Comment by dsehnal** [comment]
Mol* uses a standard file inputs: 

- Single file: `<input disabled={this.props.isDisabled} onChange={this.onChangeFile} type='file' multiple={false} accept={this.props.param.accept} />`
- Multiple files: <input disabled={this.props.isDisabled} onChange={this.onChangeFileList} type='file' multiple={true} accept={this.props.param.accept} />

The `accept` param is just comma-separated list of extensions, e.g. `.ext1,.ext2`

Perhaps you could help us troubleshoot this with a simple example of file load that works as expected?

**Comment by arose** [comment]
I don't have access to any Apple device, mobile or not, so I can't look into it.

**Comment by corredD** [comment]
seems a problem with iOS, and its seems to affect all the select file on iPhone. Also, in molstar the input file doesnt bring the default view :
![HTML attribute accept - HTML HyperText Markup Language MDN](https://github.com/molstar/molstar/assets/539461/41b61c00-11be-4645-a17f-12238c09b003)
and directly open the file browser. 

When doing the simple test with <input type="file" accept=".psf" /> the browser doesnt select anything but at least all the file can be selected. Which is not the case in molstar where no files can be selected.

**Comment by RowenaZireael** [comment]
I’m reporting this issue to Apple via the iOS18 reporting portal, let’s see if we can solve this. Or alternatively, would you consider removing the file format screening codes when user using Safari is detected (I believe that’s how they get around the issue in **_CHARMM-GUI_**)?
